grammars/css.cson

```diff
@@ -20,294 +20,1205 @@
     'patterns': [
       {
-        'include': '#string-double'
+        'begin': '{'
+        'beginCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.begin.css'
+        'end': '(})|(?<=})'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.property-list.end.css'
+        'name': 'meta.property-list.font-feature.css'
+        'patterns': [
+          {
+            # Font-feature name
+            'match': '''(?x)
+              (?: [-a-zA-Z_]    | [^\\x00-\\x7F] )   # First letter
+              (?: [-a-zA-Z0-9_] | [^\\x00-\\x7F] )*  # Remainder of identifier
+            '''
+            'name': 'variable.font-feature.css'
+          }
+          {
+            'include': '#rule-list-innards'
+          }
+        ]
+      }
+    ]
+  'functions':
+    'patterns': [
+      # Calculation
+      {
+        'begin': '(?i)(?<![\\w-])(calc)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.calc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.calc.css'
+        'patterns': [
+          {
+            'match': '[*/]|(?<=\\s|^)[-+](?=\\s|$)'
+            'name': 'keyword.operator.arithmetic.css'
+          }
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Colours
       {
-        'include': '#string-single'
+        'begin': '(?i)(?<![\\w-])(rgba?|hsla?)(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'support.function.misc.css'
+          '2':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.color.${1:/downcase}-value.css'
+        'patterns': [
+          {
+            'include': '#property-values'
+          }
+        ]
       }
+      # Gradients
       {
-        'include': '#comment-block'
+        'begin': '''(?xi) (?<![\\w-])
+          (
+            (                            # Captured for scope-name
+              (?:-webkit-|-moz-|-o-)?    # Accept prefixed/historical variants
+              (?:repeating-)?            # "Repeating"-type gradient
+              (?:linear|radial|conic)    # Shape
+            )
+            -gradient
+          )
+          (\\()
+        '''
+        'beginCaptures':
+          '1':
+            'name': 'support.function.gradient.css'
+          '3':
+            'name': 'punctuation.section.function.begin.bracket.round.css'
+        'end': '\\)'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.function.end.bracket.round.css'
+        'name': 'meta.function.${2:/downcase}.gradient.css'
```

Comment:
This could become very ugly: `meta.function.-webkit-repeating-linear.gradient.css`.  At most just include the `linear|radial|conic` section, though I'm not even sure that's needed.